hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -343,6 +356,10 @@ public Record loadRecordOrNull(Data key, boolean backup, Address callerAddress)
         return record;
     }
 
+    private long expirationTimeToTtl(long definedExpirationTime) {
+        return definedExpirationTime - System.currentTimeMillis();
```

Comment:
Shouldn't this use custom `Clock`, too, like `AbstractMapDataStore.getUserExpirationTime()`?

Reply:
We convert user expiration time to JVM based epoch as soon as we get them. Here `definedExpirationTime` is already an offset from JVM epoch.